src/tests/Uns.test.ts

```diff
@@ -752,80 +756,115 @@ describe('UNS', () => {
+          },
+        },
       });
-      const fetchSpies = mockAsyncMethods(Networking, {
-        fetch: {
-          ok: true,
-          json: () => testMeta,
+      const providerSpy = mockAsyncMethods(provider, {
+        fetchJson: {
+          jsonrpc: '2.0',
+          id: '1',
+          error: {
+            code: -32600,
+            message: 'data type size mismatch, expected 32 got 6',
+          },
         },
       });
 
       await expectResolutionErrorCode(
-        () => resolution.unhash('0x0123456789abcdef', NamingServiceName.UNS),
+        () => resolution.unhash('0xdeaddeaddead', NamingServiceName.UNS),
         ResolutionErrorCode.ServiceProviderError,
       );
-
-      expectSpyToBeCalled(unsSpies);
-      expectSpyToBeCalled(fetchSpies);
+      expectSpyToBeCalled(providerSpy);
     });
 
     it('should throw error if domain is not found', async () => {
-      const spies = mockAsyncMethods(uns.readerContract, {
-        call: new ResolutionError(ResolutionErrorCode.ServiceProviderError, {
-          providerMessage: 'execution reverted',
-        }),
-      });
-
+      const unregisteredhash = resolution.namehash(
+        'test34230131207328144694.crypto',
+      );
+      mockAPICalls('unhash', protocolLink());
       await expectResolutionErrorCode(
-        () => resolution.unhash('0xdeaddeaddead', NamingServiceName.UNS),
+        () => resolution.unhash(unregisteredhash, NamingServiceName.UNS),
         ResolutionErrorCode.UnregisteredDomain,
       );
-      expectSpyToBeCalled(spies);
+    });
+
+    skipItInLive(
+      'should throw an error if hash returned from the network is not equal to the hash provided',
+      async () => {
+        const someHash = resolution.namehash('test34230131207328144693.crypto');
+        mockAPICalls('unhash', protocolLink());
+        await expectResolutionErrorCode(
+          () => resolution.unhash(someHash, NamingServiceName.UNS),
+          ResolutionErrorCode.ServiceProviderError,
+        );
+      },
+    );
+
+    skipItInLive(
+      'getStartingBlockFromRegistry shouild return earliest for custom network',
+      async () => {
+        resolution = new Resolution({
+          sourceConfig: {
+            uns: {
+              network: 'custom',
+              url: protocolLink(),
+              proxyReaderAddress:
+                UnsConfig.networks[4].contracts.ProxyReader.address,
+            },
+          },
+        });
+        const someHash = resolution.namehash('test.coin');
+        // We need to make sure there is no mocks in the queque before we create new ones
+        nock.cleanAll();
+        mockAPICalls('unhashGetStartingBlockTest', protocolLink());
+        await expectResolutionErrorCode(
+          () => resolution.unhash(someHash, NamingServiceName.UNS),
+          ResolutionErrorCode.UnregisteredDomain,
+        );
+        // If the getStartingBlockFromRegistry function won't return "earliest" then one of the mocks will not be fired
```

Comment:
@JohnnyJumper Specify expected function call explicitly. Example: https://github.com/unstoppabledomains/resolution/blob/uns/src/tests/Resolution.test.ts#L927

Reply:
The example illustrates how to do this via the jest framework. For this test, I used nock instead and not sure how to explicitly check if the mock is called with the correct args in a different way.  In case if getStartingBlockFromRegistry returned anything except "earliest" nock will throw an error saying that mock call never happened.